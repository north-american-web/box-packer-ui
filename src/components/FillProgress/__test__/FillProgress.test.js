import React from 'react';
import 'jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';
import {FillProgress} from '../index'

afterEach(cleanup);

describe('<FillProgress/>', () => {
    it('renders and displays correctly', () => {
        const { queryByTestId, asFragment } = render(<FillProgress percent={90}/>);
        expect(queryByTestId('bar-element')).toHaveAttribute('aria-valuenow', '90');
        expect(queryByTestId('toast-element')).toBeNull();
        expect(asFragment()).toMatchSnapshot();
    });

    it('handles more than 100% correctly', () => {
        const { getByTestId, asFragment } = render(<FillProgress percent={110}/>);
        expect(getByTestId('bar-element')).toHaveAttribute('aria-valuenow', '110');
        expect(getByTestId('toast-element')).toHaveClass('toast-error');
        expect(asFragment()).toMatchSnapshot();
    })
});