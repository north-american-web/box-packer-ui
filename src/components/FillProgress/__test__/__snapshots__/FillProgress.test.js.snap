// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FillProgress/> handles more than 100% correctly 1`] = `
<DocumentFragment>
  <div
    class="bar bar-sm"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="110"
      class="bg-error"
      data-testid="bar-element"
      role="progressbar"
      style="width: 110%;"
    />
  </div>
  <div
    class="toast toast-error"
    data-testid="toast-element"
  >
    The volume of the items exceeds the volume of the container(s). The packing algorithm will not be run.
  </div>
</DocumentFragment>
`;

exports[`<FillProgress/> renders and displays correctly 1`] = `
<DocumentFragment>
  <div
    class="bar bar-sm"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="90"
      class=""
      data-testid="bar-element"
      role="progressbar"
      style="width: 90%;"
    />
  </div>
</DocumentFragment>
`;
